### PR TITLE
DDF 2716 Close streams

### PR DESCRIPTION
**Pull request by ahoffer**
#### What does this PR do?
Remediates findings from static analysis tool. The static analysis tool found instances where streams were not closed explicitly. It is believed that none of the findings posed security risks or memory leaks. In the interest of code hygiene , some instances were converted to using Java 8' try-with-resources features to ensure streams are closed explicitly.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@brendan-hofmann @AzGoalie @vinamartin 
#### Select at least one member from relevant component team(s) from below (at least one 
[Security](https://github.com/orgs/codice/teams/security)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@stustison
#### How should this be tested? (List steps with links to updated documentation)
Pass all unit and integration tests
#### Any background context you want to provide?
None.
#### What are the relevant tickets?
[DDF-2716](https://github.com/ahoffer/ddf/tree/DDF-2716)
#### Screenshots (if appropriate)

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java [hunk 1]
```diff
@@ -178,9 +178,9 @@ public Metacard transform(Map<String, Object> map) {
     }
 
     public String toMetacardXml(Metacard m) {
-        try {
-            return IOUtils.toString(catalogFramework.transform(m, "xml", null)
-                    .getInputStream());
+        try (InputStream stream = catalogFramework.transform(m, "xml", null)
+                .getInputStream()) {
+            return IOUtils.toString(stream);
         } catch (Exception e) {
```
Comment (brendan-hofmann) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java]: I don't think we should be catching `Exception`, and I'm not sure returning a blank string makes sense here.

File: catalog/transformer/catalog-transformer-video-input/src/main/java/ddf/catalog/transformer/input/video/VideoInputTransformer.java [hunk 2]
```diff
@@ -65,16 +65,17 @@ public VideoInputTransformer(MetacardType metacardType) {
...
             templates = TransformerFactory.newInstance(TransformerFactoryImpl.class.getName(),
                     this.getClass()
                             .getClassLoader())
-                    .newTemplates(new StreamSource(TikaMetadataExtractor.class.getResourceAsStream(
-                            "/metadata.xslt")));
+                    .newTemplates(new StreamSource(stream));
         } catch (TransformerConfigurationException e) {
             LOGGER.debug("Couldn't create XML transformer", e);
+        } catch (IOException e) {
+            LOGGER.debug("Could not get Tiki metadata XSLT");
```
Comment (brendan-hofmann) [inline catalog/transformer/catalog-transformer-video-input/src/main/java/ddf/catalog/transformer/input/video/VideoInputTransformer.java]: Please include the exception in the log message.

File: catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java [hunk 3]
```diff
@@ -174,16 +174,18 @@ public TikaInputTransformer(BundleContext bundleContext, MetacardType metacardTy
...
             templates =
                     TransformerFactory.newInstance(net.sf.saxon.TransformerFactoryImpl.class.getName(),
                             net.sf.saxon.TransformerFactoryImpl.class.getClassLoader())
-                            .newTemplates(new StreamSource(TikaMetadataExtractor.class.getResourceAsStream(
-                                    "/metadata.xslt")));
+                            .newTemplates(new StreamSource(stream));
         } catch (TransformerConfigurationException e) {
             LOGGER.debug("Couldn't create XML transformer", e);
+        } catch (IOException e) {
+            LOGGER.debug("Could not get Tiki metadata XSLT");
```
Comment (brendan-hofmann) [inline catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java]: Please include the exception in the log message.

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java (hunk 1 above)
Comment (ahoffer) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java]: Return null instead?

File: catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/MetacardMarshallerImpl.java [hunk 4]
```diff
@@ -257,14 +257,19 @@ private void writeAttributeToXml(PrintWriter writer, XmlPullParser parser, Attri
...
+                StandardCharsets.UTF_8.name())) {
+            XppReader source = new XppReader(inputStreamReader, parser);
+
+            // if multi-threading, cannot abstract PrintWriter to class member
+            destination = writerProvider.build(Metacard.class);
+
+            new HierarchicalStreamCopier().copy(source, destination);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to marshall metacard from content");
+            return "";
```
Comment (ahoffer) [inline catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/MetacardMarshallerImpl.java]: Almost certainly something will get jacked up down the line. It' probably better to throw a runtime exception.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java [hunk 5]
```diff
@@ -661,9 +668,13 @@ public Response createMetacard(MultipartBody multipartBody, @Context UriInfo req
...
 
-            Response.ResponseBuilder responseBuilder = Response.ok(content.getInputStream(),
-                    content.getMimeTypeValue());
-            response = responseBuilder.build();
+            try (InputStream inputStream = content.getInputStream()) {
+                Response.ResponseBuilder responseBuilder = Response.ok(inputStream,
+                        content.getMimeTypeValue());
+                response = responseBuilder.build();
+            } catch (IOException e) {
+                LOGGER.debug("Failed to get content in RESTEndpoint");
```
Comment (brendan-hofmann) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: What happens if a null response is returned from a REST method? I'm guessing jax-rs will just return a 500 or something, which might not be what we want.

Comment (ahoffer) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: The previous code didn't try to catch any exceptions from `content.getInputStream`. The exception would have bubbled up to the calling method. The only difference would be the type of exception-- `IOException` (checked, unfortunately) versus `RuntimeException`. 

If there is no clean way to use the try-with-resources, I'll probably revert my changes. The benefits of using try-with-resources has to balanced against the amount of changes that need to be make to accommodate it.

File: catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java [hunk 6]
```diff
@@ -945,10 +945,11 @@ public ResourceResponse retrieveResource(URI resourceUri,
                             requestedBytesToSkip,
                             responseBytesSkipped);
 
-                    return new ResourceResponseImpl(new ResourceImpl(new BufferedInputStream(
-                            resource.getInputStream()),
-                            resource.getMimeTypeValue(),
-                            FilenameUtils.getName(resource.getName())));
+                    try (BufferedInputStream stream = new BufferedInputStream(resource.getInputStream())) {
```
Comment (stustison) [inline catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java]: If you close this stream, the client won't be able to download the resource passed through the source.

Comment (ahoffer) [inline catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java]: That would be bad.

**Comment by shaundmorris** [comment]
@stustison are you good with the latest change on this PR?

**Pull request merged**